subsys/net/lib/sockets/socketpair.c

```diff
@@ -210,10 +219,19 @@ static void spair_delete(struct spair *spair)
  */
 static struct spair *spair_new(void)
 {
-	struct spair *spair;
+	struct spair *spair = NULL;
 	int res;
 
-#ifdef CONFIG_USERSPACE
+#ifdef CONFIG_NET_SOCKETPAIR_STATIC
```

Comment:
This search is `O(n)` but search with sys_bitarray is `O(n/64)`.. which is also `O(n)`..

I've never done a performance comparison though 🤷🏼‍♂️